module/VuFind/src/VuFind/Db/Service/AccessTokenService.php

```diff
@@ -83,7 +77,19 @@ public function getByIdAndType(
         string $type,
         bool $create = true
     ): ?AccessTokenEntityInterface {
-        return $this->accessTokenTable->getByIdAndType($id, $type, $create);
+
+        $dql = 'SELECT * '
+            . 'FROM ' . $this->getEntityClass(AccessToken::class)
+            . 'WHERE id = :id '
+            . 'AND type = :type';
+        $query = $this->entityManager->createQuery($dql);
+        $query->setParameters(['id' => $id, 'type' => $type]);
```

Comment:
Purely optional, but when you have a construction like this where array keys have the same name as the variables being assigned to them, you can shorten it with the `compact` function -- e.g. `['id' => $id, 'type' => $type]` becomes `compact('id', 'type')`.